evalml/automl/automl_search.py

```diff
@@ -378,18 +383,24 @@ def __init__(
             logger.warning(
                 "Unable to import plotly; skipping pipeline search plotting\n"
             )
-
-        if allowed_pipelines is not None and not isinstance(allowed_pipelines, list):
-            raise ValueError(
-                "Parameter allowed_pipelines must be either None or a list!"
-            )
-        if allowed_pipelines is not None and not all(
-            isinstance(p, PipelineBase) for p in allowed_pipelines
-        ):
-            raise ValueError(
-                "Every element of allowed_pipelines must an instance of PipelineBase!"
-            )
-        self.allowed_pipelines = allowed_pipelines
+        if allowed_component_graphs is not None:
+            if not isinstance(allowed_component_graphs, list):
+                raise ValueError(
+                    "Parameter allowed_component_graphs must be either None or a list!"
+                )
+            for graph in allowed_component_graphs:
+                if not isinstance(graph, (dict, ComponentGraph)):
+                    raise ValueError(
+                        "Every component graph passed must be of type dictionary or ComponentGraph!"
+                    )
+            unique_names = set()
+            for graph in allowed_component_graphs:
+                unique_names.add(list(graph.keys())[0])
```

Comment:
I'm wondering if we should change the expected type of `allowed_component_graphs` from a list of dicts to just a dict.

So from 
```python
            allowed_component_graphs=[
                {"Name_0": [dummy_classifier_estimator_class]},
                {"Name_1": [dummy_classifier_estimator_class]},
                {"Name_2": [dummy_classifier_estimator_class]},
            ],
```

to 

```python
            allowed_component_graphs={
                "Name_0": [dummy_classifier_estimator_class],
                "Name_1": [dummy_classifier_estimator_class],
                "Name_2": [dummy_classifier_estimator_class]
},
```

I see there's some duplicated code in this pr to unpack the list, e.g `list(graph.keys())[0]` and `next(iter(component_graph))` making this change may make things clearer/reduce code?

Reply:
Originally the format was set as such to support `random_seed` but since we're not passing that anymore, this would definitely be a better alternative.